storage/testdb/testdb.go

```diff
@@ -53,37 +54,47 @@ func MySQLAvailable() bool {
 }
 
 // newEmptyDB creates a new, empty database.
-func newEmptyDB(ctx context.Context) (*sql.DB, error) {
+// returns the database handle and a clean-up function, or an error.
+// The returned clean-up function should be called once the caller is finished
+// using the DB, this function may delete the underlying instance.
+func newEmptyDB(ctx context.Context) (*sql.DB, func(), error) {
 	db, err := sql.Open("mysql", *dataSourceURI)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	// Create a randomly-named database and then connect using the new name.
 	name := fmt.Sprintf("trl_%v", time.Now().UnixNano())
 
 	stmt := fmt.Sprintf("CREATE DATABASE %v", name)
 	if _, err := db.ExecContext(ctx, stmt); err != nil {
-		return nil, fmt.Errorf("error running statement %q: %v", stmt, err)
+		return nil, nil, fmt.Errorf("error running statement %q: %v", stmt, err)
 	}
 
 	db.Close()
 	db, err = sql.Open("mysql", *dataSourceURI+name)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+
+	drop := func() {
+		if _, err := db.ExecContext(ctx, "DROP DATABASE %v", name); err != nil {
```

Comment:
Do we want to use `ctx` here? If that context expires, it would prevent the cleanup, which I don't think we want.

Reply:
Yeah, I paused before using this here, but you're right - may as well just have them pass on it and be cleaner.